core/vm/lfvm/instructions.go

```diff
@@ -209,6 +255,27 @@ func opSstore(c *context) {
 
 func opSload(c *context) {
 	var top = c.stack.peek()
+
+	if c.evm.ChainConfig().IsBerlin(c.evm.Context.BlockNumber) {
+		slot := common.Hash(top.Bytes32())
+		// Check slot presence in the access list
+		if _, slotPresent := c.evm.StateDB.SlotInAccessList(c.contract.Address(), slot); !slotPresent {
+			// If the caller cannot afford the cost, this change will be rolled back
+			// If he does afford it, we can skip checking the same thing later on, during execution
+			if c.interpreter == nil {
```

Comment:
Can we also add a member function for this line, which I think is checking whether the debugger mode is active? Something like `if !c.IsShadowed() {`

Reply:
done